tests/test_taar_locale.py

```diff
@@ -130,3 +181,20 @@ def test_write_output():
     date_filename =\
         "{}{}20171106.json".format(prefix, taar_locale.LOCALE_FILE_NAME)
     assert date_filename in keys
+
+
+@mock_s3
+def test_load_amo_external_whitelist():
+
+    content = [fake_amo_sample_web_extension, fake_amo_sample_legacy]
+
+    conn = boto3.resource('s3', region_name='us-west-2')
+    conn.create_bucket(Bucket=AMO_DUMP_BUCKET)
+
+    # Store the data in the mocked bucket.
+    conn.Object(AMO_DUMP_BUCKET, key=AMO_DUMP_KEY).put(Body=json.dumps(content))
+
+    # Check that the web_extension item is still present and the legacy addon is absent.
+    white_listed_addons = taar_locale.load_amo_external_whitelist()
```

Comment:
For some reason, it looks like the library calls to boto3 are not being mocked out by moto. When running this test locally, I am getting production data.

Reply:
This is because of the caching mechanism that is implemented in `get_s3_json_content`. Either the caching mechanism is removed (it only gets run in one place), or the test harness should clear the cache directory after every test.

Reply:
Ah... nice catch! I've been going crazy testing this. Thank you!